crates/core/src/host/host_controller.rs

```diff
@@ -40,6 +41,9 @@ type HostCell = Arc<AsyncRwLock<Option<Host>>>;
 /// The registry of all running hosts.
 type Hosts = Arc<Mutex<IntMap<u64, HostCell>>>;
 
+/// A future that resolves to a [`durability::Durability`] instance.
+pub type DynDurabilityFut = BoxFuture<'static, anyhow::Result<(Arc<dyn Durability<TxData = Txdata>>, DiskSizeFn)>>;
```

Comment:
It is dyn because it needs to be wrapped around Futture, I needs to be future because we do `get_or_launch_module_host` everywhere which require this as param.

Reply:
I played around with this a bit, and I think it could work more elegantly if we have a trait like `ExternalStorage`:

```rust
pub type DurabilityAndMetrics = (Arc<dyn Durability<TxData = Txdata>>, DiskSizeFn);

// dumb name...
pub trait ExternalDurability {
    // return `None` if this replica should be in-memory only
    async fn get_durability(&self, replica_id: u64) -> anyhow::Result<Option<DurabilityAndMetrics>>;
}
```

We then store `Arc<dyn ExternalDurability>` in `HostController` instead of passing it in every method call.